### ISSUE TITLE
`Layout/LineContinuationLeadingSpace` false positive

### ISSUE THREAD

**Issue by ymap** [rubocop/rubocop#12138]
While examining the code for this rule, I noticed that the logic for detecting line continuations might mistakenly identify certain cases as such. 

In the `dstr` node type, lines ending with a backslash are flagged. However, multiline single-quoted strings, also parsed as this node type, are mistakenly flagged if they end with a backslash.

Here's a code example:

```ruby
puts '"this text is too" \
" long"'
```

---
## Expected behavior

No offences for `Layout/LineContinuationLeadingSpace` should arise.

## Actual behavior

An incorrect offence is reported as follows:

```
example.rb:2:2: C: [Correctable] Layout/LineContinuationLeadingSpace: Move leading spaces to the end of the previous line.
" long"'
 ^
```

When applying auto-correction with the `-a` option, it incorrectly alters the code's meaning as shown below:

```ruby
puts '"this text is too " \
"long"'
```

## Steps to reproduce the problem

1. Save the following code in a file named example.rb

```ruby
puts '"this text is too" \
" long"'
```

2. Run `bundle exec rubocop --only=Layout/LineContinuationLeadingSpace example.rb`

## RuboCop version

```
$ bundle exec rubocop -V
1.56.0 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.0.2) [x86_64-linux]
  - rubocop-performance 1.19.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.23.2
```

**Comment by ymap** [comment]
And I'm working on it.
